src/com/palmergames/bukkit/towny/command/TownCommand.java

```diff
@@ -498,6 +499,23 @@ public List<String> onTabComplete(CommandSender sender, Command command, String
 						default:
 							return Collections.emptyList();
 					}
+				case "trusttown":
+					switch (args.length) {
+						case 2:
+							return NameUtil.filterByStart(Arrays.asList("add", "remove", "list"), args[1]);
+						case 3:
+							if (args[1].equalsIgnoreCase("add")) {
+								List<String> townsList = getTownyStartingWith(args[2], "t");
+								townsList.removeAll(getTrustedTownsFromResidentOrThrow(player));
+								return townsList;

```

Comment:
I believe this still needs to be `return NameUtil.filterByStart(townsList, args[2]);`

Please test it out.

Reply:
This method returns the correct expected input in-game but it may not be the most efficient, the method you propose would remove the earlier requested functionality regarding the trusttown command filtering for towns which are already added.
![filtering2](https://user-images.githubusercontent.com/90512509/199692942-99874593-6838-4841-bb08-749a6eb7cb80.png)
![filtering1](https://user-images.githubusercontent.com/90512509/199692950-65703c55-22e5-44a0-a7c3-2f6e16e615e2.png)

Reply:
I understand, but I think without the filterByStart its not going to filter out names as they are typed no? You would need to test using names that are more different than Test1 and Test2.

Reply:
Ah I see what you mean now, getTownyStartingWith will already filter names as typed using a string pass as 'arg'.
![image](https://user-images.githubusercontent.com/90512509/199996898-18d8c171-61f1-4bea-8bae-e848a5893a48.png)
![image](https://user-images.githubusercontent.com/90512509/199996913-8f861cae-cbc4-48fe-9ce7-dadd466042fb.png)
![image](https://user-images.githubusercontent.com/90512509/199996938-77f56a93-1c23-4630-88bb-08f91687b2f2.png)
Here are some examples using this method.